src/components/Identity/createIdSyncs.js

```diff
@@ -61,40 +61,38 @@ const createProcessor = (config, logger, cookieJar) => destinations => {
       )
     );
 
-  if (idSyncs.length) {
-    return fireDestinations({
-      logger,
-      destinations: idSyncs
-    }).then(result => {
-      const nowInHours = Math.round(
-        convertTimes(MILLISECOND, HOUR, new Date().getTime())
-      );
+  return fireDestinations({
+    logger,
+    destinations: idSyncs
+  }).then(result => {
+    const nowInHours = Math.round(
+      convertTimes(MILLISECOND, HOUR, new Date().getTime())
+    );
 
-      result.succeeded.forEach(idSync => {
-        const ttlInHours = Math.round(
-          convertTimes(
-            MINUTE,
-            HOUR,
-            idSync.ttlMinutes || DEFAULT_ID_SYNC_TTL_MINUTES
-          )
-        );
-
-        controlObject[idSync.id] = nowInHours + ttlInHours;
-      });
-
-      setControlObject(controlObject, cookieJar);
-
-      cookieJar.set(
-        ID_SYNC_TIMESTAMP,
-        (
-          Math.round(convertTimes(MILLISECOND, HOUR, new Date().getTime())) +
-          convertTimes(DAY, HOUR, 7)
-        ).toString(36)
+    result.succeeded.forEach(idSync => {
+      const ttlInHours = Math.round(
+        convertTimes(
+          MINUTE,
+          HOUR,
+          idSync.ttlMinutes || DEFAULT_ID_SYNC_TTL_MINUTES
+        )
       );
+
+      controlObject[idSync.id] = nowInHours + ttlInHours;
     });
-  }
 
-  return Promise.resolve();
+    setControlObject(controlObject, cookieJar);
+
+    cookieJar.set(
+      ID_SYNC_TIMESTAMP,
+      (
+        Math.round(convertTimes(MILLISECOND, HOUR, new Date().getTime())) +
+        convertTimes(DAY, HOUR, 7)
+      ).toString(36)
+    );
+
+    return Promise.resolve(result);
```

Comment:
You're already inside a `.then`, so there's not need to return a promise. Just do `return result;`

Reply:
If the result has failures, shouldn't we reject the promise provided to the customer since at least part of the ID syncing actually failed? I would think if all ID syncs were successful, we should resolve the promise with no resolution value. If any of the ID syncs failed, we should reject the promise with an error that probably contains some details about which ID syncs failed (maybe list out the IDs in the error message).

Reply:
I'm testing to see if all id syncs were processed, whether successful or failed, in order to know when to remove the iframe.